src/hu_stuff.c

```diff
@@ -1153,25 +1153,51 @@ static void HU_widget_build_sttime(void)
   int offset = 0;
   extern int time_scale;
 
-  if (time_scale != 100)
+  if (gameaction == ga_completed)
+    plr->btuse_tics = 0;
+
+  if ((hud_level_time & HUD_WIDGET_HUD     && !automapactive) ||
```

Comment:
It still stinks to me to perform the exact same checks again here, but I couldn't find a different solution either.

Reply:
Maybe we can use XOR to make it shorter:
`if (hud_level_time && hud_level_time ^ (automapactive + 1))`.

I think that's not what you want, though.

Reply:
No, sorry, but this is ugly and unintuitive. And I'm not even sure if it's semantically correct.